dynamiq/nodes/tools/basetool.py

```diff
@@ -0,0 +1,45 @@
+from abc import ABC, abstractmethod
+from typing import Any, Literal
+
+from pydantic import BaseModel, Field, ValidationInfo, field_validator
+
+from dynamiq.nodes import NodeGroup
+from dynamiq.runnables import RunnableConfig
+
+
+class DefaultInputSchema(BaseModel, ABC):
+    input: str
+
+
+class ToolMixin:
```

Comment:
Why do you use tool mixing and don't apply changes to `Node`?
It seems make sense to update `Node` and add `input_schema` to base class now (Optional for now).
Also makes sense to define `validate_input_schema` with validation Node if input_schema exists.